main.tf

```
@@ -1 +1,375 @@
+  iam_role_external_id            = module.lacework_eks_audit_iam_role.external_id
+  cross_account_iam_role_name     = "${var.prefix}-ca-${random_id.uniq.hex}"
+  sns_name                        = "${var.prefix}${random_id.uniq.hex}"
+  firehose_iam_role_name          = "${var.prefix}-fh-${random_id.uniq.hex}"
+  firehose_policy_name            = "${var.prefix}-fh-policy-${random_id.uniq.hex}"
+  firehose_delivery_stream_name   = "${var.prefix}${random_id.uniq.hex}"
+  eks_cw_iam_role_name            = "${var.prefix}-cw-${random_id.uniq.hex}"
+  cw_iam_policy_name              = "${var.prefix}-cw-policy-${random_id.uniq.hex}"
+  cloudwatch_permission_resources = "arn:aws:logs:*:${data.aws_caller_identity.current.account_id}:log-group:/aws/eks/*:*"
+  log_regions                     = [for region in var.cloudwatch_regions : "logs.${region}.amazonaws.com"]
+  cluster_names                   = var.no_cw_subscription_filter ? [] : var.cluster_names
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+resource "aws_sns_topic" "eks_sns_topic" {
+  name = local.sns_name
+}
+
+resource "aws_sns_topic_policy" "eks_sns_topic_policy" {
+  arn = aws_sns_topic.eks_sns_topic.arn
+
+  policy = data.aws_iam_policy_document.eks_sns_topic_policy.json
+}
+
+# we need the identity of the caller to get their account_id
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "eks_sns_topic_policy" {
+
+  statement {
+    sid    = "AllowLaceworkToSubscribe"
+    effect = "Allow"
+    principals {
+      identifiers = ["arn:aws:iam::${var.lacework_aws_account_id}:root"]
+      type        = "AWS"
+    }
+    actions = [
+      "sns:Subscribe"
+    ]
+    resources = [
+      aws_sns_topic.eks_sns_topic.arn
+    ]
+  }
+
+  statement {
+    sid = "AllowS3ToPublish"
+    actions = [
+      "sns:Publish",
+    ]
+    resources = [
+      aws_sns_topic.eks_sns_topic.arn
+    ]
+    condition {
+      test     = "StringEquals"
+      variable = "AWS:SourceAccount"
+
+      values = [
+        data.aws_caller_identity.current.account_id,
+      ]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "AWS:SourceArn"
+
+      values = [
+        "arn:aws:s3:*:*:${local.bucket_name}",
+      ]
+    }
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_s3_bucket" "eks_audit_log_bucket" {
+  bucket        = local.bucket_name
+  force_destroy = var.bucket_force_destroy
+  tags          = var.tags
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "eks_audit_log_bucket_lifecycle_config" {
+  bucket = aws_s3_bucket.eks_audit_log_bucket.id
+
+  rule {
+    id = "eks_audit_log_expiration"
+    expiration {
+      days = var.bucket_lifecycle_expiration_days
+    }
+    status = "Enabled"
+  }
+}
+
+// v4 s3 bucket changes
+resource "aws_s3_bucket_versioning" "export__versioning" {
```

Did we mean to have multiple underscores?

```suggestion
resource "aws_s3_bucket_versioning" "export_versioning" {
```